src/TTextEdit.cpp

```diff
@@ -1784,6 +1805,433 @@ int TTextEdit::getRowCount()
+                                         byteToLuaCodeOrChar(&utf8Bytes[3])));
+                }
+                utf8Index += 4;
+                break;
+            case 3:
+                if (includeThisCodePoint) {
+                    utf8Indexes.append(QStringLiteral("<th><center>%1</center></th><td><center>%2</center></td><td><center>%3</center></td>")
+                                       .arg(QString::number(utf8Index), QString::number(utf8Index+1), QString::number(utf8Index+2)));
+                    utf8Vals.append(QStringLiteral("<td><center>0x%1</center></td><td><center>0x%2</center></td><td><center>0x%3</center></td>")
+                                    .arg(static_cast<quint8>(utf8Bytes[0]), 2, 16, zero)
+                            .arg(static_cast<quint8>(utf8Bytes[1]), 2, 16, zero)
+                            .arg(static_cast<quint8>(utf8Bytes[2]), 2, 16, zero));
+                    luaCodes.append(QStringLiteral("<td><center>%1</center></td><td><center>%2</center></td><td><center>%3</center></td>")
+                                    .arg(byteToLuaCodeOrChar(&utf8Bytes[0]),
+                                         byteToLuaCodeOrChar(&utf8Bytes[1]),
+                                         byteToLuaCodeOrChar(&utf8Bytes[2])));
+                }
+                utf8Index += 3;
+                break;
+            case 2:
+                if (includeThisCodePoint) {
+                    utf8Indexes.append(QStringLiteral("<th><center>%1</center></th><td><center>%2</center></td>")
+                                       .arg(QString::number(utf8Index), QString::number(utf8Index+1)));
+                    utf8Vals.append(QStringLiteral("<td><center>0x%1</center></td><td><center>0x%2</center></td>")
+                                    .arg(static_cast<quint8>(utf8Bytes[0]), 2, 16, zero)
+                            .arg(static_cast<quint8>(utf8Bytes[1]), 2, 16, zero));
+                    luaCodes.append(QStringLiteral("<td><center>%1</center></td><td><center>%2</center></td>")
+                                    .arg(byteToLuaCodeOrChar(&utf8Bytes[0]),
+                                         byteToLuaCodeOrChar(&utf8Bytes[1])));
+                }
+                utf8Index += 2;
+                break;
+            default:
+                if (includeThisCodePoint) {
+                    utf8Indexes.append(QStringLiteral("<th><center>%1</center></th>").arg(QString::number(utf8Index)));
+                    utf8Vals.append(QStringLiteral("<td><center>0x%1</center></td>").arg(static_cast<quint8>(utf8Bytes[0]), 2, 16, zero));
+                    luaCodes.append(QStringLiteral("<td><center>%1</center></td>").arg(byteToLuaCodeOrChar(&utf8Bytes[0])));
+                }
+                ++utf8Index;
+            }
+
+            if (includeThisCodePoint) {
+                ++rowItems;
+            }
+        }
+
+        if (rowItems > rowLimit) {
+            if (isFirstRow) {
```

Comment:
Selecting multiple lines breaks the table:

![image](https://user-images.githubusercontent.com/110988/47546735-ddd1ce80-d8f2-11e8-8bc9-51f69776cd4f.png)

Reply:
That `isFirstRow` is what is used to only show those *sidings* (horizontal headings? :stuck_out_tongue_winking_eye:) on the first row of the table.  It'd be interested to check the code that works out what part of the `TBuffer` to analyse is deciding in that situation - perhaps I broke something when I revised it to only consider the selected part of the first line... :cry:

Reply:
That shouldn't be happening - I'll have to take another look...

Reply:
I can't reproduce this - have you got a replay with that text in it?